op-bindings/bindgen/remote_handlers.go

```diff
@@ -0,0 +1,282 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/etherscan"
+)
+
+type contractData struct {
+	abi          string
+	deployedBin  string
+	deploymentTx etherscan.TxInfo
+}
+
+func (generator *bindGenGeneratorRemote) standardHandler(contractMetadata *remoteContractMetadata) error {
+	fetchedData, err := generator.fetchContractData(contractMetadata.Verified, "eth", contractMetadata.Deployments["eth"])
+	if err != nil {
+		return err
+	}
+
+	contractMetadata.DeployedBin = fetchedData.deployedBin
+
+	// If ABI was explicitly provided by config, don't overwrite
+	if contractMetadata.Abi == "" {
+		contractMetadata.Abi = fetchedData.abi
+	} else if fetchedData.abi != "" && contractMetadata.Abi != fetchedData.abi {
+		generator.logger.Warn("The given contract ABI differs from what was fetched from Etherscan", "contract", contractMetadata.Name)
```

Comment:
curious if you think this should cause an error or if a warning is sufficient

Reply:
This should be an error IMO - could lead to silent failures.